apps/emqx_durable_storage/src/emqx_ds_optimistic_tx.erl

```diff
@@ -574,8 +571,29 @@ finalize_add_pending(_, D = #d{n_items = N, max_items = MaxItems}, Actions) when
     handle_flush(D, Actions);
 finalize_add_pending(?idle, D = #d{idle_flush_interval = IdleInterval}, Actions) ->
     %% Schedule early flush if the shard is idle:
-    Timeout = {timeout, IdleInterval, ?timeout_flush},
+    Timeout = {state_timeout, IdleInterval, ?timeout_flush},
```

Comment:
The original problem which made `emqx_mgmt_api_clients_SUITE.t_list_clients_v2_limit` was the following:
* Several (e.g.) clients disconnect almost at once
* They try to clear will timers and do this within `idle_flush_interval`
* Two tx-es arrive into this process
* One moves the process `idle` -> `pending` and schedules early flush
* The second arrives while into `pending` state and cancels early flush (`{timeout, ..., ...}` actions are cancelled by any event, is this intentional? 🤔) 
* The flush happens only after `state_timeout` which is significant: 500ms

So in the test, some clients were alive quite long after disconnect command, causing flakyness.